test/e2e/Tests/Fixtures/FunctionAppFixture.cs

```diff
@@ -0,0 +1,165 @@
+            {
+                // kill existing func processes
+                this._logger.LogInformation("Shutting down any running functions hosts..");
+                FixtureHelpers.KillExistingProcessesMatchingName("func");
+
+                // start functions process
+                this._logger.LogInformation($"Starting functions host for {Constants.FunctionAppCollectionName}...");
+
+                string rootDir = Path.GetFullPath(@"../../../../../../");
+                string e2eAppBinPath = Path.Combine(rootDir, @"test/e2e/Apps/BasicDotNetIsolated/bin");
+                string? e2eHostJson = Directory.GetFiles(e2eAppBinPath, "host.json", SearchOption.AllDirectories).FirstOrDefault();
+
+                if (e2eHostJson == null)
+                {
+                    throw new InvalidOperationException($"Could not find a built worker app under '{e2eAppBinPath}'");
+                }
+
+                string? e2eAppPath = Path.GetDirectoryName(e2eHostJson);
+
+                if (e2eAppPath == null)
+                {
+                    throw new InvalidOperationException($"Located host.json for app at {e2eHostJson} but could not resolve the app base directory");
+                }
+
+                this._funcProcess = FixtureHelpers.GetFuncHostProcess(e2eAppPath);
+                string workingDir = this._funcProcess.StartInfo.WorkingDirectory;
+                this._logger.LogInformation($"  Working dir: '${workingDir}' Exists: '{Directory.Exists(workingDir)}'");
+                string fileName = this._funcProcess.StartInfo.FileName;
+                this._logger.LogInformation($"  File name:   '${fileName}' Exists: '{File.Exists(fileName)}'");
+
+                //TODO: This may be added back if we want cosmos tests
+                //await CosmosDBHelpers.TryCreateDocumentCollectionsAsync(_logger);
+
+                //TODO: WORKER ATTACH ISSUES
+                //      Abandoning this attach method for now - It seems like Debugger.Launch() from the app can't detect the running VS instance.
+                //      Not sure if this is because VS is the parent process, or because it is already attached to testhost.exe, but for now we 
+                //      will rely on manual attach. Some possible solution with DTE might exist but for now, it relies on a specific VS version
+                //if (Debugger.IsAttached)
+                //{
+                //    _funcProcess.StartInfo.EnvironmentVariables["DURABLE_ATTACH_DEBUGGER"] = "True";
+                //}
+
+                FixtureHelpers.StartProcessWithLogging(this._funcProcess, this._logger);
+
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    // ensure child processes are cleaned up
+                    _jobObjectRegistry = new JobObjectRegistry();
+                    _jobObjectRegistry.Register(this._funcProcess);
+                }
+
+                var httpClient = new HttpClient();
```

Comment:
Minor: It's important to dispose `HttpClient` objects when we're doing with them to avoid resource leaks. A simple `using` on this line will ensure that this object gets cleaned up at the end of this method invocation.
```suggestion
                using var httpClient = new HttpClient();
```